analyses/filter-mutation-frequencies-tables/filter-mutation-frequencies-tables.Rmd

```diff
@@ -0,0 +1,192 @@
+
+
+## Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('filter-mutation-frequencies-tables.Rmd', clean = TRUE)"
+```
+_This assumes you are in the modules directory of the repository, OpenPedCan-analysis/analyses/filter-mutation-frequency-tables._
+
+## Setup
+
+Load libraries
+```{r}
+# R analysis packages
+suppressPackageStartupMessages(library(data.table))
+suppressPackageStartupMessages(library(tidyverse))
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+
+Set up directories. 
+```{r}
+# directories for input and output files
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+analyses_dir <- file.path(root_dir, "analyses")
+module_dir <- file.path(analyses_dir, "filter-mutation-frequencies-tables")
+snv_results_dir <- file.path(analyses_dir,  "snv-frequencies/results")
+cnv_results_dir <- file.path(analyses_dir,  "cnv-frequencies/results")
+fusion_results_dir <- file.path(analyses_dir,  "fusion-frequencies/results")
+results_dir <- file.path(module_dir, "results")
+
+# Create results folder if it doesn't exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+## Get Ensembl gene identifiers
+
+Load Gencode gtf file
+```{r}
+# gtf column names
+col_names <- c("seqname", "source", "feature", "start", "end", "score", "strand", "frame", "attribute")
+
+# read file
+gene_code_annot <- 
+  data.table::fread(file.path(module_dir, "input", "gencode.v38.annotation.gtf.gz"),
+                    col.names = col_names, skip=5, 
+                    showProgress = FALSE)
+```
+
+Get Ensembl identifiers
+```{r}
+# filter for gene rows and get ids
+ensg_ids <- gene_code_annot %>% 
+  dplyr::filter(feature == "gene") %>% 
+  dplyr::mutate(gene_id = stringr::str_extract(attribute, "ENSG\\d+")) %>% 
+  pull(gene_id) %>% 
+  unique()
+
+# Remove gencode dataframe
+rm(gene_code_annot)
+```
+ 
+## Filter mutation frequencies tables
+
+Function to filter for required Ensembl gene identifiers and write to file
+```{r}
+
+filter_freq_table <- function(freq_file, ensg_ids) {
+  
+  # Read gene level table 
+  freq_table <-  data.table::fread(freq_file, showProgress = FALSE) 
+  
+  # Write filtered TSV file
+  file_name <- basename(freq_file)
+  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
+  data.table::fwrite(file.path(results_dir, file_name), sep = "\t", compress = "auto")
+
+  # Write filtered JSON file
+  json_file <- paste(unlist(str_split(file_name, "\\."))[1], "json", sep = ".")
+  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
+    jsonlite::write_json(file.path(results_dir,  json_file))  
+
+return(freq_table)
+}
+```
+
+#### Gene level SNV frequencies
+```{r}
+# Remove Ensembl gene identifiers
+gene_level_snv <- filter_freq_table(
+  file.path(snv_results_dir, "gene-level-snv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 
+
+# Ensembl gene identifiers removed
+gene_level_snv %>% dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
```

Comment:
Looks like the checks for what gets filtered out are not working for all tables. From the code it looks like you are asking which ENSG IDs are not in v38 AFTER the filtering step , which should be none if the filtering step works
Instead. I am interested in seeing which ones get filtered out so we should ask which ENSG IDs are not in v38 BEFORE filtering?
Additionally, could you please also output the `Kids_First_Biospecimen_ID` associated with those filtered ENSG IDs so that if necessary, we would have a record of what we might need to re-align manually? 
Thanks!

Reply:
@runjin326, please carefully look at the filtering function, `filter_freq_table()` in `filter-mutation-frequencies-tables.Rmd`. It returns an unfiltered data frame of the frequencies tables. All filtering is done in the intermediate sections of the function code and written to file without altering the load data frame. Others there would have been no results of filtered ENSGs. 

How do we output `Kids_First_Biospecimen_ID` associated with those filtered `ENSGs`? The frequencies tables don't have a list of `sample IDs` associated with mutation `ENSGs`. Instead, I have included the associated gene symbols.

Reply:
@ewafula, I see - sorry about the confusion!
So what I was thinking was - now that you have ENSGs, can you go back to the original table (i.e., `fusion-putative-oncogenic.tsv` for fusions, `consensus_wgs_plus_cnvkit_wxs.tsv.gz` for CNV and `snv-consensus-plus-hotspots.maf.tsv.gz` for SNV), find out those ENSGs and output the `Kids_First_Biospecimen_ID`s associated with those ENSGs - basically trying to figure out how many samples are affected and who are they.
Additionally, now that I am thinking about this, maybe also use `histologies.tsv`, filter to those samples, select cohorts and do a `table` call to get a summary of what cohorts they belong to.

Reply:
Sure, @runjin326. Will do.

Reply:
> Sure, @runjin326. Will do.

@runjin326, wrote dropped ENSG IDs to output files in addition to printing the tables out in the notebook. Too many associated Kids_First_Biospecimen_IDs for one to patiently page through.